packages/main/src/DatePicker.js

```diff
@@ -503,6 +525,87 @@ class DatePicker extends UI5Element {
 				this._toggleAndFocusInput();
 			}
 		}
+
+		if (this.isOpen()) {
+			return;
+		}
+
+		if (isPageUpShiftCtrl(event)) {
+			event.preventDefault();
+			this._changeDateValue(true, true, false, false);
+		} else if (isPageUpShift(event)) {
+			event.preventDefault();
+			this._changeDateValue(true, false, true, false);
+		} else if (isPageUp(event)) {
+			event.preventDefault();
+			this._changeDateValue(true, false, false, true);
+		}
+
+		if (isPageDownShiftCtrl(event)) {
+			event.preventDefault();
+			this._changeDateValue(false, true, false, false);
+		} else if (isPageDownShift(event)) {
+			event.preventDefault();
+			this._changeDateValue(false, false, true, false);
+		} else if (isPageDown(event)) {
+			event.preventDefault();
+			this._changeDateValue(false, false, false, true);
+		}
+	}
+
+	/**
+	 * Adds or extracts a given number of measuring units from the "dateValue" property value
+	 *
+	 * @param {boolean} years indicates that the measuring unit is in years
+	 * @param {boolean} months indicates that the measuring unit is in months
+	 * @param {boolean} days indicates that the measuring unit is in days
+	 * @param {boolean} forward if true indicates addition
+	 * @param {int} step number of measuring units to substract or add defaults ot 1
+	 */
+	_changeDateValue(forward, years, months, days, step = 1) {
+		let date = this.dateValue;
+
+		if (!date) {
+			return;
+		}
+
+		const oldDate = new Date(date.getTime());
+		const incrementStep = forward ? step : -step;
+
+		if (incrementStep === 0) {
```

Comment:
I guess we can check, because in future the same method could be used for different interactions and a 'step'
parameter value different than defafult value.